diff --git a/cartographer_ros/configuration_files/assets_writer_backpack_3d.lua b/cartographer_ros/configuration_files/assets_writer_backpack_3d.lua
index 43c2bd9..5e02fd6 100644
--- a/cartographer_ros/configuration_files/assets_writer_backpack_3d.lua
+++ b/cartographer_ros/configuration_files/assets_writer_backpack_3d.lua
@@ -48,6 +48,46 @@ options = {
       transform = XZ_TRANSFORM,
     },
 
+    -- We now use the intensities to color our points. We apply a linear
+    -- transform to clamp our intensity values into [0, 255] and then use this
+    -- value for RGB of our points. Every stage in the pipeline after this now
+    -- receives colored points.
+    --
+    -- We write xrays again. These now use geometry and the intensities to
+    -- color pixels - they look quite similar, just a little lighter.
+    {
+      action = "intensity_to_color",
+      min_intensity = 0.,
+      max_intensity = 4095.,
+    },
+
+    {
+      action = "write_xray_image",
+      voxel_size = VOXEL_SIZE,
+      filename = "xray_yz_all_intensity",
+      transform = YZ_TRANSFORM,
+    },
+    {
+      action = "write_xray_image",
+      voxel_size = VOXEL_SIZE,
+      filename = "xray_xy_all_intensity",
+      transform = XY_TRANSFORM,
+    },
+    {
+      action = "write_xray_image",
+      voxel_size = VOXEL_SIZE,
+      filename = "xray_xz_all_intensity",
+      transform = XZ_TRANSFORM,
+    },
+
+    -- We also write a PLY file at this stage, because gray points look good.
+    -- The points in the PLY can be visualized using
+    -- https://github.com/googlecartographer/point_cloud_viewer.
+    {
+      action = "write_ply",
+      filename = "points.ply",
+    },
+
     -- Now we recolor our points by frame and write another batch of X-Rays. It
     -- is visible in them what was seen by the horizontal and the vertical
     -- laser.
diff --git a/cartographer_ros/configuration_files/backpack_3d.lua b/cartographer_ros/configuration_files/backpack_3d.lua
index 123543f..6f9da7e 100644
--- a/cartographer_ros/configuration_files/backpack_3d.lua
+++ b/cartographer_ros/configuration_files/backpack_3d.lua
@@ -27,7 +27,7 @@ options = {
   num_laser_scans = 0,
   num_multi_echo_laser_scans = 0,
   num_subdivisions_per_laser_scan = 1,
-  num_point_clouds = 2,
+  num_point_clouds = 1,
   lookup_transform_timeout_sec = 0.2,
   submap_publish_period_sec = 0.3,
   pose_publish_period_sec = 5e-3,
@@ -40,7 +40,7 @@ options = {
 TRAJECTORY_BUILDER_3D.num_accumulated_range_data = 160
 
 MAP_BUILDER.use_trajectory_builder_3d = true
-MAP_BUILDER.num_background_threads = 7
+MAP_BUILDER.num_background_threads = 6
 POSE_GRAPH.optimization_problem.huber_scale = 5e2
 POSE_GRAPH.optimize_every_n_nodes = 320
 POSE_GRAPH.constraint_builder.sampling_ratio = 0.03
diff --git a/cartographer_ros/launch/backpack_3d.launch b/cartographer_ros/launch/backpack_3d.launch
index 52dea59..3ffda10 100644
--- a/cartographer_ros/launch/backpack_3d.launch
+++ b/cartographer_ros/launch/backpack_3d.launch
@@ -26,6 +26,7 @@
           -configuration_directory $(find cartographer_ros)/configuration_files
           -configuration_basename backpack_3d.lua"
       output="screen">
+    <remap from="points2" to="horizontal_laser_3d" />
     <remap from="points2_1" to="horizontal_laser_3d" />
     <remap from="points2_2" to="vertical_laser_3d" />
   </node>
diff --git a/cartographer_ros/launch/offline_backpack_3d.launch b/cartographer_ros/launch/offline_backpack_3d.launch
index e9d7ba8..a684914 100644
--- a/cartographer_ros/launch/offline_backpack_3d.launch
+++ b/cartographer_ros/launch/offline_backpack_3d.launch
@@ -27,6 +27,7 @@
           -urdf_filename $(find cartographer_ros)/urdf/backpack_3d.urdf
           -bag_filenames $(arg bag_filenames)"
       output="screen">
+    <remap from="points2" to="horizontal_laser_3d" />
     <remap from="points2_1" to="horizontal_laser_3d" />
     <remap from="points2_2" to="vertical_laser_3d" />
   </node>
